quantestpy/assertion/get_ctrl_val.py

```diff
@@ -0,0 +1,95 @@
+import copy
+import unittest
+
+import numpy as np
+
+from quantestpy import PauliCircuit
+from quantestpy.exceptions import QuantestPyAssertionError, QuantestPyError
+
+ut_test_case = unittest.TestCase()
+
+
+def _get_qubit_idx_to_ctrl_val_for_given_val_in_select_reg(
+        val_in_select_reg: str,
+        pc: PauliCircuit,
+        select_reg: list,
+        ancilla_reg: list) -> dict:
+
+    # init ancilla reg to 0
+    pc.set_qubit_value(ancilla_reg, [0] * len(ancilla_reg))
+    # init select reg to val_in_select_reg
+    pc.set_qubit_value(select_reg, [int(i) for i in val_in_select_reg])
+
+    # get ctrl vals for all ops on syst reg
```

Comment:
When I name a dictionary, I use the convention "{key}_to_{value}" since a dictionary has the key-value structure. But other conventions are welcome. What convention is `ctrl_dict_of_qubit_idx_and_qubit_val` based on? Note that it is important to use the common naming convention everywhere (i.e. not only this dictionary but also all the other dictionaries). Please let us discuss this on a meeting.

Reply:
```
ctrl_qubit_idx_to_ctrl_qubit_val = {idx: [] for idx in select_reg + ancilla_reg}
for i, gate in enumerate(pc._gates):
        ctrl_qubit_idx = gate["control_qubit"]
        ctrl_qubit_val = pc._qubit_value[ctrl_qubit_idx].tolist()

        for j, idx in enumerate(ctrl_qubit_idx):
             if idx in ctrl_qubit_idx_to_ctrl_qubit_val.keys():
                 ctrl_qubit_idx_to_ctrl_qubit_val[qubit_idx].append(ctrl_qubit_val[j])

         pc._execute_i_th_gate(i)
```